contracts/injectorFactoryV2.sol

```diff
@@ -35,14 +45,22 @@ contract ChildChainGaugeInjectorV2Factory {
         uint256 minWaitPeriodSeconds,
         address injectTokenAddress,
         uint256 maxInjectionAmount,
-        address owner
+        address owner,
+        bool registerUpkeep
     ) external returns (address) {
         address injector = Clones.clone(implementation);
         ChildChainGaugeInjectorV2(injector).initialize(
             owner, keeperAddresses, minWaitPeriodSeconds, injectTokenAddress, maxInjectionAmount
         );
         emit InjectorCreated(injector, keeperAddresses, injectTokenAddress, owner);
         deployedInjectors.push(injector);
+
+        if (registerUpkeep) {
+            uint256 upkeepId = _registerUpkeep(injector, owner);
+            address[] memory keepers = new address[](1);
+            keepers[0] = CL_REGISTRY.getForwarder(upkeepId);
```

Comment:
Doesn't this get setup in another tx, not sure you can getForwarder for a new upkeep in the same tx as you deploy it.  Have you tested this?